File: Day 1/8/rotate_an_array.c
#include <stdio.h>

void rotateArray(int arr[], int n, int k) {
    k = k % n;
    int rotatedArray[n];
    
    for (int i = 0; i < n; i++) {
        rotatedArray[(i + k) % n] = arr[i];
    }
    
    for (int i = 0; i < n; i++) {
        printf("%d ", rotatedArray[i]);
    }
    printf("\n");
}

int main() {
    int n, k;
    
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    
    int arr[n];
    printf("Enter the elements of the array: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    
    printf("Enter the number of rotations: ");
    scanf("%d", &k);
    
    rotateArray(arr, n, k);
    return 0;
}

File: Day 1/8/rotateAnArray.js
const rotateAnArray = (arr, k) => {
    const n = arr.length;
    k = k % n;
    return arr.slice(-k).concat(arr.slice(0, n - k));
}

const rotateAnArrayAlgorithmically = (arr, k) => {
    const n = arr.length;
    k = k % n;
    let rotatedArray = new Array(n);
    
    for (let i = 0; i < n; i++) {
        rotatedArray[(i + k) % n] = arr[i];
    }
    
    return rotatedArray;
}

console.log(rotateAnArray([1, 2, 3, 4, 5, 6, 7], 10));
console.log(rotateAnArrayAlgorithmically([1, 2, 3, 4, 5, 6, 7], 10));